clippy_lints/src/methods/mod.rs

```diff
@@ -3087,6 +3088,37 @@ fn lint_search_is_some<'tcx>(
             span_lint(cx, SEARCH_IS_SOME, expr.span, &msg);
         }
     }
+    // lint if `find()` is called by `String` or `&str`
+    else if search_method == "find" {
+        let is_string_or_str_slice = |e| {
+            let self_ty = cx.typeck_results().expr_ty(e).peel_refs();
+            if is_type_diagnostic_item(cx, self_ty, sym!(string_type)) {
+                true
+            } else {
+                *self_ty.kind() == ty::Str
+            }
+        };
+        if_chain! {
+            if is_string_or_str_slice(&search_args[0]);
+            if is_string_or_str_slice(&search_args[1]);
+            then {
+                let msg = "called `is_some()` after calling `find()` \
+                    on a string. This is more succinctly expressed by calling \
+                    `contains()`.".to_string();
```

Comment:
```suggestion
                    `contains()`";
```

Reply:
Thanks!

Reply:
@flip1995: Did you want to remove the period after `contains()`?  I think this is causing the test to fail and I will need to run `update-all-references.sh` again to update the `stderr` file if so.  

However, I just squashed commits on my local copy to make a rebase easier, so the commits will be changed if I do this (not sure if that's a problem or not).